### DISCUSSION TITLE
CC2531 Device Limit

### DISCUSSION THREAD

**Discussion by markhotchkiss** [Koenkk/zigbee2mqtt#6004] [General]
Hi everyone,

I am pretty new to all this, so please go easy!

I have the CC2531 running as a coordinator and I have just recently realised that there is a device limit if you don’t have a router! (Lesson 1, read the docs...)

I’ve tried to pair another 3 devices and it seems that they won’t pair, I assume because I have hit the device limit.

After some looking around and reading the docs, it seems that buying a router is the way to go to get more devices....

However...in order to get one of my new devices paired, I removed some of my other devices to make space and I have found myself in a situation where I now can’t pair the new devices or get the old devices back again!

Therefore my questions are:

1. Can I reset the coordinator to “forget” all devices and then repair the ones I want?
2. If I add a router, will this fail to join because the “coordinator” has reached its device limit?

any help would be greatly appreciated.

**Comment by Koenkk** [comment]
1. Only way is to reflash it
2. When the coordinator has reached it limit and there are no other routers with free space available nothing happens.

**Reply by markhotchkiss** [reply]
OK, thanks....

I have made some headway this afternoon by using my Philips Hue bulbs as routers.  It would seem that I can remove some devices, add the bulbs as routers then re-add my devices again and that works.

The only thing now, is that there are obviously still "devices" paired with the CC2531 which I don't have any more, so can't un-pair them! Shame the only way around this is to reflash, as I don't have the debugger kit.  Also, reading the docs, the impression I got was that if you reflash, you **don't** have to repair the devices.  So, will flashing it work?

**Reply by markhotchkiss** [reply]
oh and thank you very much for the help, it is greatly appreciated!!!

**Reply by Koenkk** [reply]
Reflashing indeed doesn't require repairing.

**Reply by maddocche** [reply]
Sorry to bring this up again, but I'm having the same problem where I can't pair old devices that was paired before and then deleted (to be specific, these are IKEA Tradfri Bulbs). I re-flashed the CC2531 and as you said it doesn't require repairing. So, how can I make the coordinator/Z2M forget these devices to pair them again? Thx!

**Reply by markhotchkiss** [reply]
I reflashed and it did seem to work...

if you need to, you can make it so all devices would have to re-pair by changing the network key.

you can force Zigbee2Mqtt to do this by adding something to the configuration.yaml file as documented on this page at the bottom: https://www.zigbee2mqtt.io/how_tos/how_to_secure_network.html
